app/grandchallenge/components/backends/base.py

```diff
@@ -370,19 +370,7 @@ def _provision_inputs(self, *, input_civs, input_prefixes):
                 civ=civ, input_prefixes=input_prefixes
             )
 
-            if civ.image:
-                self._copy_input_file(src=civ.image_file, dest_key=key)
-            elif civ.file:
-                self._copy_input_file(src=civ.file, dest_key=key)
-            else:
-                with io.BytesIO() as f:
-                    f.write(json.dumps(civ.value).encode("utf-8"))
-                    f.seek(0)
-                    self._s3_client.upload_fileobj(
-                        Fileobj=f,
-                        Bucket=settings.COMPONENTS_INPUT_BUCKET_NAME,
-                        Key=key,
-                    )
```

Comment:
@koopmant watch out for things like this! I have refactored this one to look at the super kind.